src/components/DataDimension/DataElementSelector.js

```diff
@@ -80,112 +122,123 @@ DisaggregationSelector.propTypes = {
     onChange: PropTypes.func.isRequired,
 }
 
-const DataElementSelector = ({
-    displayNameProp,
-    selectedItems = [],
-    onSelect,
-}) => {
+const DataElementSelector = ({ displayNameProp, onSelect }) => {
     const dataEngine = useDataEngine()
-    const rootRef = useRef()
-    const didMountRef = useRef(false)
-    const [searchInput, setSearchInput] = useState()
-    const [filter, setFilter] = useState({
-        dataType: DIMENSION_TYPE_DATA_ELEMENT,
-        subGroup: TOTALS,
-    })
+
+    const [searchTerm, setSearchTerm] = useState('')
+    const [group, setGroup] = useState(DIMENSION_TYPE_ALL)
+    const [subGroup, setSubGroup] = useState(TOTALS)
     const [options, setOptions] = useState([])
     const [loading, setLoading] = useState(true)
-    const [page, setPage] = useState(0)
-    const [hasNextPage, setHasNextPage] = useState(true)
-    const debouncedSearchTerm = useDebounce(searchInput, 200)
 
-    if (selectedItems.length) {
-        // FIXME: temporarily removes lint errors
-        console.log(loading + setFilter())
-    }
+    const rootRef = useRef()
+    const hasNextPageRef = useRef(false)
+    const searchTermRef = useRef(searchTerm)
+    const pageRef = useRef(0)
+    const filterRef = useRef({
+        dataType: DIMENSION_TYPE_DATA_ELEMENT,
+        group,
+        subGroup,
+    })
 
-    const fetchItems = useCallback(
-        async (searchTerm) => {
+    const fetchData = async (scrollToTop) => {
+        try {
             setLoading(true)
+
             const result = await apiFetchOptions({
                 dataEngine,
                 nameProp: displayNameProp,
-                page,
-                filter,
-                searchTerm,
+                filter: filterRef.current,
+                searchTerm: searchTermRef.current,
+                page: pageRef.current,
             })
-            const newOptions = []
-            result.dimensionItems?.forEach((item) => {
-                newOptions.push({
+
+            // XXX also check for length?
```

Comment:
needed?

Reply:
No, `dimensionItems` should always be an array when defined.